platform/core/src/DICOMSR/dataExchange.js

```diff
@@ -78,4 +82,112 @@ const storeMeasurements = async (measurementData, filter, server) => {
   }
 };
 
-export { retrieveMeasurements, storeMeasurements };
+const downloadReport = measurementData => {
+  const srDataset = generateReport(measurementData);
+  const reportBlob = dcmjs.data.datasetToBlob(srDataset);
+
+  //Create a URL for the binary.
+  var objectUrl = URL.createObjectURL(reportBlob);
+  window.location.assign(objectUrl);
+};
+
+const generateReport = measurementData => {
+  const ids = measurementData.map(md => md.id);
+  const filteredToolState = _getFilteredCornerstoneToolState(ids);
+
+  const report = MeasurementReport.generateReport(
+    filteredToolState,
+    cornerstone.metaData
+  );
+
+  return report.dataset;
+};
+
+const storeMeasurements = async (measurementData, dataSource) => {
+  // TODO -> Eventually use the measurements directly and not the dcmjs adapter,
+  // But it is good enough for now whilst we only have cornerstone as a datasource.
+  log.info('[DICOMSR] storeMeasurements');
+
+  if (!dataSource || !dataSource.store || !dataSource.store.dicom) {
+    log.error('[DICOMSR] datasource has no dataSource.store.dicom endpoint!');
+    return Promise.reject({});
+  }
+
+  const naturalizedReport = generateReport(measurementData);
+  const { StudyInstanceUID } = naturalizedReport;
+
+  try {
+    await dataSource.store.dicom(naturalizedReport);
+
+    if (StudyInstanceUID) {
+      studies.deleteStudyMetadataPromise(StudyInstanceUID);
+    }
+
+    return {
+      message: 'Measurements saved successfully',
+    };
+  } catch (error) {
+    log.error(
+      `[DICOMSR] Error while saving the measurements: ${error.message}`
+    );
+    throw new Error('Error while saving the measurements.');
```

Comment:
If you haven't already, separate issue calling out UI notification for success/fail.
Outside of sprint, blocked by design/user/func requirements

Reply:
https://precisionmetrics.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=OHIF&modal=detail&selectedIssue=OHIF-177